packages/core/src/http/requestBuilder.ts

```diff
@@ -540,9 +555,12 @@ export class DefaultRequestBuilder<BaseUrlParamType, AuthParams>
         this._validateResponse &&
         (response.statusCode < 200 || response.statusCode >= 300)
       ) {
-        throw new this._apiErrorFactory(
+        if (typeof this._apiErrorFactory?.message === 'undefined') {
+          this._apiErrorFactory.message = `Response status code was not ok: ${response.statusCode}.`;
+        }
+        throw new this._apiErrorFactory.apiErrorCtor(
           context,
-          `Response status code was not ok: ${response.statusCode}.`
+          this._apiErrorFactory.message
```

Comment:
Is discarding `Response status code was not ok: ${response.statusCode}.` altogether okay?

Reply:
Its not being discarded, being set the same message incase of it being undefined.